src/System.Windows.Extensions/src/System/Security/Cryptography/X509Certificates/X509Utils.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+using System.Security.Cryptography.X509Certificates;
+
+namespace System.Security.Cryptography.X509Certificates
+{
+    internal static class X509Utils
+    {
+        internal const uint CERT_STORE_ENUM_ARCHIVED_FLAG = 0x00000200;
+        internal const uint CERT_STORE_CREATE_NEW_FLAG = 0x00002000;
+
+        internal static SafeCertContextHandle GetCertContext(X509Certificate2 certificate)
+        {
+            SafeCertContextHandle safeCertContext = Interop.Crypt32.CertDuplicateCertificateContext(certificate.Handle);
+            GC.KeepAlive(certificate);
```

Comment:
Could CertDuplicateCertificateContext be defined to take a SafeHandle instead of an IntPtr?  That would be safer and wouldn't require this GC.KeepAlive.

Reply:
X509Certificate/X509Certificate2 don't expose SafeHandle members, just the IntPtr one.